elfloader-tool/src/binaries/efi/efi_fputc.c

```diff
@@ -19,14 +19,16 @@ int efi_fputc(int c, UNUSED FILE *stream)
     /* obtain a pointer to EFI output console */
     efi_simple_text_output_protocol_t * efi_console_out = get_efi_con_out();
 
-    /* EFI_SIMPLE_TEXT_OUTPUT_PROTOCOL only accepts unicode chars */
-    uint16_t char_to_print[2] = { c, 0 };
-
+    /* Send '\r' (CR) before every '\n' (LF). */
     if (char_to_print[0] == '\n')
```

Comment:
Now that char_to_print is defined on line 31 I think this will produce a compilation error.  Otherwise everything else looks good to merge.

Reply:
good catch, will change this. Actually, I've been looking though the code where efi_fputc() is used, but I could not find any reference. Seem this does not ride on the usual code path of arch_write_char(() calling _fputc(), but is something special?
I have another patch pending that changes arch_write_char() to print the CR in case of LF. Then the code duplication can be removed from all the _fputc() implementations and they can stick to just printing whatever gets thrown into them.

Reply:
Yea it looks like it was added when EFI support was initially added but isn't used anywhere